error/s2n_errno.h

```diff
@@ -333,6 +340,21 @@ extern __thread const char *s2n_debug_str;
  */
 #define S2N_IFF(a, b) (!!(a) == !!(b))
 
+/**
+ * These macros are used to specify code contracts in CBMC proofs.
+ */
+#ifdef CBMC
```

Comment:
I'm not sure s2n_errno is the correct location for this. I made utils/s2n_ensure.h for this kind of stuff

Reply:
I only placed them here because of the `S2n_IMPLIES` and `S2N_IFF`, so I guess I should move those to `utils/s2n_ensure.h` as well.

Reply:
Yeah those don't really belong here either :)

Reply:
Implemented as suggested.